SplashScreen/app/src/main/java/com/example/android/splashscreen/MainActivity.kt

```diff
@@ -15,19 +17,26 @@ import com.example.android.splashscreen.databinding.MainActivityBinding
 
 /**
  * Shows the app content that is commonly used in all of [DefaultActivity], [AnimatedActivity], and
- * [CustomActivity]. This also handles the custom dark mode.
+ * [CustomActivity]. This also handles the custom dark mode on API level 31 and above.
  */
 abstract class MainActivity : AppCompatActivity() {
 
     private val viewModel: MainViewModel by viewModels()
 
+    protected lateinit var installedSplashScreen: SplashScreen
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        // Set up 'core-splashscreen' to handle the splash screen in a backward compatible manner.
```

Comment:
I've noticed that it's often safer to call this before super.onCreate()

Reply:
Done